pyvista/plotting/affine_widget.py

```diff
@@ -0,0 +1,372 @@
+        Direction of the ray.
+    plane_point : ndarray
+        A point on the plane.
+    normal : ndarray
+        Normal to the plane.
+
+    Returns
+    -------
+    ndarray
+        Intersection point.
+    """
+    t_value = np.dot(normal, (plane_point - start_point)) / np.dot(normal, direction)
+    return start_point + t_value * direction
+
+
+class AffineWidget3D:
+    """3D affine transform widget.
+
+    This widget allows interactive transformations including translation and
+    rotation using the left mouse button.
+
+    Parameters
+    ----------
+    plotter : pyvista.Plotter
+        The plotter object.
+    actor : pyvista.Actor
+        The actor to which the widget is attached to.
+    origin : sequence[float], optional
+        Origin of the widget. Default is the center of the main actor.
+    start : bool, default: True
+        If True, start the widget immediately.
+    scale : float, default: 0.15
+        Scale factor for the widget relative to the length of the actor.
+    line_radius : float, default: 0.02
+        Relative radius of the lines composing the widget.
+    always_visible : bool, default: True
+        Make the widget always visible. Setting this to ``False`` will cause
+        the widget geometry to be hidden by other actors in the plotter.
+    axes_colors : tuple[ColorLike], optional
+        Uses the theme by default. Configure the individual axis colors by
+        modifying either the theme with ``pv.global_theme.axes.x_color =
+        <COLOR>`` or setting this with a ``tuple`` as in ``('r', 'g', 'b')``.
+    callback : callable, optional
+        Call this method when releasing the left mouse button. It is passed the
+        ``user_matrix`` of the actor.
+
+    Notes
+    -----
+    After interacting with the actor, the transform will be stored within
+    :attr:`pyvista.Actor.user_matrix` but will not be applied to the
+    dataset. Use this matrix in conjunction with
+    :func:`pyvista.DataSetFilters.transform` to transform the dataset.
+
+    Requires VTK >= v9.2
+
+    Examples
+    --------
+    Create the affine widget outside of the plotter and add it.
+
+    >>> import pyvista as pv
+    >>> pl = pv.Plotter()
+    >>> actor = pl.add_mesh(pv.Sphere())
+    >>> widget = pv.AffineWidget3D(pl, actor)
+    >>> pl.show()
+
+    Access the transform from the actor.
+
+    >>> actor.user_matrix
+    array([[1., 0., 0., 0.],
+           [0., 1., 0., 0.],
+           [0., 0., 1., 0.],
+           [0., 0., 0., 1.]])
+
+    """
+
+    def __init__(
+        self,
+        plotter,
+        actor,
+        origin=None,
+        start=True,
+        scale=0.15,
+        line_radius=0.02,
+        always_visible=True,
+        axes_colors=None,
+        callback=None,
+    ):
+        """Initialize the widget."""
+        # needs VTK v9.2.0 due to the hardware picker
+        if pv.vtk_version_info < (9, 2):
+            VTKVersionError('AfflineWidget3D requires VTK v9.2.0 or newer.')
```

Comment:
```suggestion
        if pv.vtk_version_info < (9, 2):  # pragma: no cover
            VTKVersionError('AfflineWidget3D requires VTK v9.2.0 or newer.')
```

Reply:
Should we rather have a test that covers this?